graph/api/src/main/java/org/jboss/windup/graph/model/ApplicationArchiveModel.java

```diff
@@ -3,7 +3,7 @@
 import com.tinkerpop.frames.modules.typedgraph.TypeValue;
 
 @TypeValue(ApplicationArchiveModel.TYPE)
-public interface ApplicationArchiveModel extends ApplicationModel, ArchiveModel
+public interface ApplicationArchiveModel extends ArchiveModel
```

Comment:
Why would this no longer extend ApplicationModel?

Reply:
Actually, I see why, but this all seems a bit strange now. It seems to me that ApplicationModel is a marker interface to indicate that something is an application.

Previously, (filemodel instanceof ApplicationModel) == true for cases where the file was an application.

After this change, that relationship is broken as you have redefined the meaning of "ApplicationModel".

Reply:
Yes, it's an marker interface. But the name is ambiguous, so I am creating ApplicationProjectModel, which marks a ProjectModel that is an app, and a ApplicationInputPathModel, which marks what is now FileModel instanceof ApplicationModel.

Reply:
Note that a directory may also be an application, while not being an archive.